ci-operator/step-registry/ipi/conf/capability/additionalcaps/ipi-conf-capability-additionalcaps-commands.sh

```diff
@@ -81,32 +81,48 @@ fi
 # shellcheck disable=SC2206
 vcurrent_capabilities_array=(${vcurrent_capabilities})
 echo "vcurrent_capabilities: ${vcurrent_capabilities_array[*]}"
-selected_capability_index=$((RANDOM % ${#vcurrent_capabilities_array[@]}))
-selected_capability="${vcurrent_capabilities_array[$selected_capability_index]}"
-echo "Selected capability to be disabled: ${selected_capability}"
 
 enabled_capabilities=${vcurrent_capabilities}
-case "${selected_capability}" in
-"MachineAPI")
-    echo "WARNING: MachineAPI is selected, but it requires on IPI, so no capability to be disabled!"
-    ;;
-# To be updated once OCP 4.16 is released
-"CloudCredential")
-    if [[ "${CLUSTER_TYPE}" =~ ^packet.*$|^equinix.*$ ]]; then
-        enabled_capabilities=${enabled_capabilities/${selected_capability}}
-    else
-        echo "WARNING: non-BareMetal platforms require CCO for OCP 4.15, so no capability to be disabled!"
+additional_caps_from_config=$(yq-go r "${SHARED_DIR}/install-config.yaml" "capabilities.additionalEnabledCapabilities[*]")
+selected_capability=""
+while [[ -z "${selected_capability}" ]]; do
+    selected_capability_index=$((RANDOM % ${#vcurrent_capabilities_array[@]}))
+    selected_capability="${vcurrent_capabilities_array[$selected_capability_index]}"
+    # If selected to be disabled cap has already been set to additionalEnabledCapabilities in install-config, should not be disabled
+    if [[ "${additional_caps_from_config}" =~ ${selected_capability} ]]; then
+        echo "WARNING: selected cap ${selected_capability} is already configured in field additionalEnabledCapabilities in install-config, unable to be disabled!"
+        selected_capability=""
+        continue
     fi
-    ;;
-# Disable marketplace if OperatorLifecycleManager is selected to bo disabled
-"OperatorLifecycleManager")
-    echo "Capability 'marketplace' depends on Capability 'OperatorLifecycleManager', so disable marketplace along with OperatorLifecycleManager"
-    enabled_capabilities=${enabled_capabilities/${selected_capability}}
-    enabled_capabilities=${enabled_capabilities/"marketplace"}
-    ;;
-*)
-    enabled_capabilities=${enabled_capabilities/${selected_capability}}
-esac
+    case "${selected_capability}" in
+    "MachineAPI")
+        echo "WARNING: MachineAPI is selected, but it requires on IPI, could not be disabled!"
+        selected_capability=""
+        ;;
+    # To be updated once OCP 4.16 is released
+    "CloudCredential")
+        if [[ "${CLUSTER_TYPE}" =~ ^packet.*$|^equinix.*$ ]]; then
+            enabled_capabilities=${enabled_capabilities/${selected_capability}}
+        else
+            echo "WARNING: non-BareMetal platforms require CCO for OCP 4.15, could not be disabled!"
+            selected_capability=""
+        fi
+        ;;
+    # Disable marketplace if OperatorLifecycleManager is selected to bo disabled
+    "OperatorLifecycleManager")
```

Comment:
How about introducing the similar array like https://github.com/openshift/release/pull/49364#discussion_r1508421490

then remove this branch, but move it to the next branch.
```
    *)
       enabled_capabilities=${enabled_capabilities/${selected_capability}}
        for key in ${!dependency_caps[@]}; do
            if [[ "${dependency_caps[$key]}" == "${selected_capability}" ]]; then
                echo "Capability '$key' depends on Capability '${dependency_caps[$key]}', so disable $key along with ${dependency_caps[$key]}"
                enabled_capabilities=${enabled_capabilities/"$key"}
            fi
        done
```

Reply:
updated